src/contrib/cluster/Akka.DistributedData/ReplicatorSettings.cs

```diff
@@ -52,35 +52,32 @@ public static ReplicatorSettings Create(Config config)
 
             if (durableKeys.Count != 0)
             {
-                Type durableStoreType;
                 if (string.IsNullOrEmpty(durableStoreTypeName))
                 {
                     throw new ArgumentException($"`akka.cluster.distributed-data.durable.store-actor-class` must be set when `akka.cluster.distributed-data.durable.keys` have been configured.");
                 }
 
-                durableStoreType = Type.GetType(durableStoreTypeName);
+                var durableStoreType = Type.GetType(durableStoreTypeName);
                 if (durableStoreType is null)
                 {
-                    throw new ArgumentException($"`akka.cluster.distributed-data.durable.store-actor-class` is set to an invalid class {durableStoreType}.");
+                    throw new ArgumentException($"`akka.cluster.distributed-data.durable.store-actor-class` is set to an invalid class {durableStoreTypeName}.");
                 }
                 durableStoreProps = Props.Create(durableStoreType, durableConfig).WithDispatcher(dispatcher);
             }
 
-            // TODO: This constructor call fails when these fields are not populated inside the Config object:
-            // TODO: `pruning-marker-time-to-live` key depends on Config.GetTimeSpan() to return a TimeSpan.Zero default.
             return new ReplicatorSettings(
-                role: config.GetString("role"),
-                gossipInterval: config.GetTimeSpan("gossip-interval"),
-                notifySubscribersInterval: config.GetTimeSpan("notify-subscribers-interval"),
-                maxDeltaElements: config.GetInt("max-delta-elements"),
+                role: config.GetString("role", string.Empty),
```

Comment:
These are just code clean-up, while providing default values for most of the settings.